analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantNullCheck.CSharp9.Fixed.cs

```diff
@@ -155,7 +155,7 @@
 result = (a is null && a is string); // Compliant - rule ConditionEvaluatesToConstant should raise issue here (to be consistent with the non-C#9 tests)
 x = m switch
 {
-    string s2 and null => s2, // Compliant - rule ConditionEvaluatesToConstant will raise issue here
+    string s2 and null => s2, // Error [CS8502] - Compliant - rule ConditionEvaluatesToConstant will raise issue here
```

Comment:
Can you give the compiler message. Is `ConditionEvaluatesToConstant` is adopted too?

Reply:
> [CS8510]: The pattern is unreachable. It has already been handled by a previous arm of the switch expression or it is impossible to match.

`ConditionEvaluatesToConstant` did not need any changes.